### PR TITLE
Added initial version of query-mapper

### PR DESCRIPTION
**Pull request by ottovw**
includes:
- Value Mapper
- Value List Mapper

### CODE REVIEW

File: src/query-mapper/query-mapper.interface.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    constructor(readonly mappers: ParameterMapper[]) { }
+
+    map(query: Object): MappedQuery {
+        if (query == null) {
+            return null
+        }
+
+        const mappedQuery = {}
+        this.mappers.forEach(mapper => {
```
Comment (edgarmueller) [inline src/query-mapper/query-mapper.interface.ts:22]: minor: `reduce` would be more readable

File: src/query-mapper/query-mapper.spec.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    const mapper = new TestMapper([
+      new ValueMapper('foo')
+    ])
+
+    const result = mapper.map(query)
+
+    expect(result).toHaveProperty('foo', 'bar')
+  });
+
+  it('value mapper should return nothing if no value', () => {
```
Comment (edgarmueller) [inline src/query-mapper/query-mapper.spec.ts:50]: no value or not existing?

File: src/query-mapper/mapper/value.mapper.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { AbstractParameterMapper } from "./parameter-mapper.interface"
+
+// todo type check?!
+export class ValueMapper extends AbstractParameterMapper {
+    doMap(parameter: any) {
+        if (parameter == null) {
+            return null
+        }
+        if (typeof parameter == 'object' || Array.isArray(parameter)) {
```
Comment (edgarmueller) [inline src/query-mapper/mapper/value.mapper.ts:9]: second condition isn't needed, I think

**Review by edgarmueller — APPROVED** [review]

File: src/query-mapper/mapper/value.mapper.ts (hunk 3 above)
Comment (ottovw) [reply src/query-mapper/mapper/value.mapper.ts:9]: this is done to accommodate that req.query params sometime comes as array or as object. didn't understand the logic. but this accommodates both cases.

**Pull request merged**